src/components/message/message-card/MessageCard.tsx

```diff
@@ -110,44 +119,71 @@ const MessageCard = ({ message, viewType, setViewType }: Props) => {
 		bookmarksStore.toggleMessagePin(message);
 	}, [bookmarksStore.toggleMessagePin]);
 
-	const addMessagesToExport = React.useCallback(
-		() => messagesStore.exportStore.addMessageToExport(message),
-		[messagesStore.exportStore.addMessageToExport],
-	);
-
 	const isExported = messagesStore.exportStore.isExported(message);
 
-	const rootClass = createBemBlock(
-		'message-card-wrapper',
-		isAttached ? 'attached' : null,
-		isBookmarked ? 'pinned' : null,
-		isHighlighted ? 'highlighted' : null,
-		isSoftFiltered ? 'soft-filtered' : null,
-		messagesStore.exportStore.isExport ? 'export-mode' : null,
-		isExported ? 'exported' : null,
-	);
+	const messageCardBaseProps: MessageCardBaseProps = {
+		message,
+		viewTypeConfig: viewTypesConfig,
+		hoverMessage,
+		unhoverMessage,
+		addMessagesToExport,
+		isHighlighted,
+		isSoftFiltered,
+		isExported,
+		isExpanded: props.isExpanded,
+		isExport: messagesStore.exportStore.isExport,
+		isBookmarked,
+		isAttached,
+		toogleMessagePin,
+		sortOrderItems,
+	};
 
 	return (
-		<div className={rootClass} onClick={addMessagesToExport}>
-			<MessageCardBase
-				message={message}
-				viewType={viewType}
-				setViewType={setViewType}
-				hoverMessage={hoverMessage}
-				unhoverMessage={unhoverMessage}
-				isBookmarked={isBookmarked}
-				isAttached={isAttached}
-				toogleMessagePin={toogleMessagePin}
-				sortOrderItems={sortOrderItems}
+		<div className='messages-list__item'>
+			<MessageCardBase {...messageCardBaseProps} />
+			<MessageExpandButton
+				isExpanded={props.isExpanded}
+				setExpanded={props.setIsExpanded}
+				parsedMessages={message.parsedMessages}
 			/>
 		</div>
 	);
-};
+});
 
 const RecoverableMessageCard = (props: OwnProps) => {
-	const viewTypesStore = useMessagesViewTypesStore();
-	const { viewType, setViewType } = viewTypesStore.getSavedViewType(props.message);
-	return <MessageCard {...props} viewType={viewType} setViewType={setViewType} />;
+	const viewTypesStore = useMessagesViewTypeStore();
+	const { getSavedViewType } = viewTypesStore;
+	const viewTypesConfig = props.message.parsedMessages
+		? ([
+				...props.message.parsedMessages.map(parsedMessage => {
+					return {
+						viewType: getSavedViewType(props.message, parsedMessage.id).viewType,
+						setViewType: getSavedViewType(props.message, parsedMessage.id).setViewType,
+					};
+				}),
+				{
+					viewType: getSavedViewType(props.message).viewType,
+					setViewType: getSavedViewType(props.message).setViewType,
+				},
+		  ] as MessageViewTypeConfig[])
```

Comment:
i think this as MessageViewTypeConfig can be omited
and if type was required better way do it would be pass a type to .map like so
`.map<MessageViewTypeConfig>()`